File: ShopApp.WebUI/Controllers/HomeController.fs
namespace ShopApp.WebUI.Controllers

open System
open System.Collections.Generic
open System.Linq
open System.Threading.Tasks
open Microsoft.AspNetCore.Mvc

type HomeController () =
    inherit Controller()

    member this.Index () =
        this.View()

    member this.Privacy () =
        this.View()

    member this.Error () =
        this.View();

File: ShopApp.UI/Controller/HomeController.cs
using Microsoft.AspNetCore.Mvc;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using ShopApp.Business.Abstract;
using ShopApp.UI.Models;

namespace ShopApp.UI.Controller
{
    public class HomeController : Microsoft.AspNetCore.Mvc.Controller
    {
        private IProductService _productService;
       

        public HomeController(IProductService productService)
        {
            _productService = productService;
        }

        public IActionResult Index()
        {
            return View(new ProductListModel()
            {
                Products=_productService.GetAll()
            });
        }
    }
}
